tests/helper-functions.test.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+function set_up_before_script() {
+    mock "fzf" "echo 0.999.0"
+    source bin/git-forgit
+}
+
+# @data_provider provider_diff_lines
+function test_forgit_get_files_from_diff_line() {
+    local -r input="$1"
+    shift
+    local -r expected=("$@")
+    local -a actual=()
+    local i
+
+    while IFS= read -r line; do
+        actual+=( "$line" )
+    done < <( echo -n "$input" | _forgit_get_files_from_diff_line | xargs -0 -n 1 echo )
+
+    # Compare array sizes
+    assert_same "${#expected[@]}" "${#actual[@]}"
+
+    # Compare array elements
+    for i in "${!expected[@]}"; do
+        echo "${actual[i]}"
+        assert_same "${expected[i]}" "${actual[i]}"
+    done
+}
+
+function provider_diff_lines() {
+    data_set "[A]     newfile" "newfile"
+    data_set "[D]     oldfile with spaces" "oldfile with spaces"
+    data_set "[R100]  file  ->  another file" "file" "another file"
```

Comment:
We should probably add a few extra data sets for esoteric file names. Files with backslashes or line breaks come to my mind.

Reply:
Good idea! I have added a testcase with a tab-containing file in its escaped form, just like it is output from `git diff --name-status` (which is the input to `_forgit_get_files_from_diff_line`, after being manipulated slightly, but that would be a different unit test). The unit test for this function passes. However, I noticed that the preview window and on pressing enter the file diff is not shown (most probably due to what you already said in #465 concerning git's output and input formats). This should not be handled by this function, though, so IMO this unit test is complete.

Reply:
Thanks for the update. Is there any reason not to add a filename with a literal backslash in it. I think this would be a perfect candidate to prevent future regressions.

Reply:
We could totally do that. It's not the expected (and actual) type of input for this particular function, though, because it consumes output of `git status`, and that prints, as you already stated, escaped file names. That's why I don't think that's a useful test case for this function.
For other functions we will definitely need literal tabs in the test data, but here I don't see why that would make sense. Maybe this function was not a good choice for the first test case. :-/

Reply:
Ah, makes sense, no need to add it then. Thanks for the explanation :+1: